src/awkward/_v2/_slicing.py

```diff
@@ -413,15 +434,12 @@ def prepare_tuple_bool_to_int(item):
 
         else:
             return ak._v2.contents.IndexedOptionArray(
-                item.index,
-                item.content,
-                identifier=item.identifier,
-                parameters=item.parameters,
+                item.index, prepare_tuple_bool_to_int(item.content)
             )
 
     elif isinstance(item, ak._v2.contents.NumpyArray):
         assert item.data.shape == (item.length,)
-        return item.data
+        return item
```

Comment:
This one resolves a type inconsistency. The function should always return a Content, rather than sometimes returning an array. It's been handled at the call site as well.